thrust/thrust/iterator/counting_iterator.h

```diff
@@ -53,7 +53,7 @@
 THRUST_NAMESPACE_BEGIN
 
 // forward declaration of counting_iterator
-template <typename Incrementable, typename System, typename Traversal, typename Difference>
+template <typename Incrementable, typename System, typename Traversal, typename Difference, typename Step>
```

Comment:
I believe we should use the empty base class optimization here and inherit from `Step` We could also generalize `__mdspan_ebco` so that we do not store unneeded information

Reply:
Doesn't EBO get us in trouble with MSVC + nvcc and having different sizes in host/device code?

Reply:
Implemented using EBO